### PR TITLE
login_call_resetPassword

### LINKED ISSUE #833: 🔧 [Refactor] Implement Login Call loginViaEmailVerificationCode on Apollo
## 🔧 Refactor ticket

Implement Login Call loginViaEmailVerificationCode on Apollo

### LINKED ISSUE #836: 🔧 [Refactor] Implement Login Call sendEmail on Apollo
## 🔧 Refactor ticket

Implement Login Call sendEmail on Apollo

### LINKED ISSUE #837: 🔧 [Refactor] Implement Login Call resetPassword on Apollo
## 🔧 Refactor ticket

Implement Login Call resetPassword on Apollo

### PR DESCRIPTION
**Pull request by ulfgebhardt**
## 🍰 Pullrequest

Implement the existing calls:
- resetpassword
- checkEmail
- loginViaEmailVerificationCode
on Apollo.

Multiple processes will be unified and simplified:
- On Register -> send EMail -> set Password & Activate -> Login
- On Forgot Password -> send EMail -> set Password -> Login
- On NoPassword but Activated -> send Email -> set Password & Activate -> Login (legacy)

Following Calls will be implemented:
- `requestSetPassword`
- `setPassword`

Furthermore the two-step-call process will be a one-step-call process.
> Before: EMail Link -> Call for Login on open page-> Call for Reset Password/Activate Account on submit
> After: EMail Link -> Call for Reset Password/Activate Account on submit

### Issues

- fixes #836  
- fixes #837 
- fixes #833 
- includes #1131

### Todo

- [x] remove `checkEmail` & `loginViaEmailVerificationCode` calls
- [x] combine `checkEmail` & `loginViaEmailVerificationCode` with `resetPassword` calls into single `setPassword` call
- [x] integrate with `createUser` process
- [x] integrate with `loginUser` process for legacy reasons & not activated
- [ ] frontend implementations @Elweyn

### CODE REVIEW

**Comment by Elweyn** [comment]
frontend implementations TODOs
- [ ] Withdraw password fields of registration form
- [ ] Make a view that will handle (set password, Forgot password & loginViaEmailVerificationCode) same as thx.vue

File: backend/.env.dist [hunk 1]
```diff
@@ -2,8 +2,6 @@ PORT=4000
 JWT_SECRET=secret123
 JWT_EXPIRES_IN=10m
 GRAPHIQL=false
-LOGIN_API_URL=http://login-server:1201/
-COMMUNITY_API_URL=http://nginx/api/
```
Comment (Mogge) [inline backend/.env.dist:6]: image

**Comment by ulfgebhardt** [comment]
image

i skipped the whole resetPassword test since it completely changed - I think we need to rewrite it once we done with this

**Comment by ulfgebhardt** [comment]
I'd like to merge this and do the frontend part in another PR to reduce complexity. This means we reduce coverage requirement for the frontend to 84% again, since the whole check/reset-Password Component has to change and therefore no test is working anylonger on that. Whats your verdict @Mogge ?

**Comment by Elweyn** [comment]
@ulfgebhardt I changed the frontend on this branch!
- In case a User has not activated his email/account and try to login => get's redirected to /reset/login that shows a message to check his email to activate his account!
- In case a User sends a setPassword and get's an optin expired => Users get's redirected to the resetPassword page with a new message that says optin has expired want a new?

**Review by Elweyn — APPROVED** [review]

**Pull request merged**